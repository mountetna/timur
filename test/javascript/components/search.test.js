import React from 'react';
import {Provider} from 'react-redux';
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {mockStore, stubUrl, cleanStubs} from '../helpers';
import Search, {
  Search as RawSearchComponent
} from '../../../lib/client/jsx/components/search/search';
import SelectInput from '../../../lib/client/jsx/components/inputs/select_input';

const models = {
  monster: {template: require('../fixtures/template_monster.json')},
  labor: {template: require('../fixtures/template_labor.json')},
  project: {template: require('../fixtures/template_project.json')}
};

describe('Search', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      magma: {models},
      search: {
        model_name: 'Labors',
        pages: {}
      }
    });
  });

  afterEach(cleanStubs);

  it('renders', () => {
    global.CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <Search />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
