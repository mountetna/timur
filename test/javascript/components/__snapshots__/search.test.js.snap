// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<div
  id="search"
>
  <div
    className="control"
  >
    <div
      className="query"
    >
      <div
        className="table-select"
      >
        <span
          className="label"
        >
          Search table
        </span>
        <select
          defaultValue=""
          name="model"
          onChange={[Function]}
        >
          <option
            disabled={false}
            value=""
          >
             --- 
          </option>
          <option
            value={0}
          >
            labor
          </option>
          <option
            value={1}
          >
            monster
          </option>
          <option
            value={2}
          >
            project
          </option>
        </select>
      </div>
      <div
        className="page-select"
      >
        <span
          className="label"
        >
          Page size
        </span>
        <select
          defaultValue={0}
          onChange={[Function]}
        >
          <option
            disabled={true}
            value=""
          >
             --- 
          </option>
          <option
            value={0}
          >
            10
          </option>
          <option
            value={1}
          >
            25
          </option>
          <option
            value={2}
          >
            50
          </option>
          <option
            value={3}
          >
            200
          </option>
        </select>
      </div>
      <div
        className="query-options"
      />
      <input
        className="button disabled"
        disabled={true}
        id="search-pg-search-btn"
        onClick={[Function]}
        type="button"
        value="Search"
      />
      <input
        className="button disabled"
        disabled={true}
        id="search-pg-tsv-btn"
        onClick={[Function]}
        type="button"
        value="â†¯ TSV"
      />
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="table"
    />
  </div>
</div>
`;
